src/main/java/com/p14n/postevent/PostgresDebeziumConnector.java

```diff
@@ -0,0 +1,55 @@
+package com.p14n.postevent;
+
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.engine.format.Json;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+public class PostgresDebeziumConnector {
+    private final Properties properties;
+    private Consumer<ChangeEvent<String, String>> changeEventConsumer;
+    private DebeziumEngine<ChangeEvent<String, String>> engine;
+
+    public PostgresDebeziumConnector(Properties properties) {
+        this.properties = properties;
+    }
+
+    public void setChangeEventConsumer(Consumer<ChangeEvent<String, String>> consumer) {
+        this.changeEventConsumer = consumer;
+    }
+
+    public void start() {
+        if (changeEventConsumer == null) {
+            throw new IllegalStateException("Change event consumer must be set before starting the engine");
+        }
+
+        this.engine = io.debezium.engine.DebeziumEngine.create(Json.class)
+                .using(properties)
+                .notifying(changeEventConsumer)
+                .build();
+
+        // Start the engine in a separate thread
+        Thread thread = new Thread(() -> {
+            try {
+                engine.run();
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to start Debezium engine", e);
+            }
+        });
+        thread.setName("debezium-engine");
+        thread.start();
```

Comment:
Pass in an ExecutorService argument to start

Also engine.run does not throw any exceptions